hdf/src/hdf.h

```diff
@@ -145,6 +145,17 @@ typedef intn (*hdf_termfunc_t)(void);   /* termination function typedef */
 
 /* .................................................................. */
 
+/* Publically useful macros for tests/examples.  */
+
+/* Used to make certain a checked value _is_not_ a specific value.
```

Comment:
if(ret == val) an error message is printed.  I still think the comment should read "If true, meaning that th checked value matches the specific value, print an error message."